### PR TITLE
[WIP] chore: upgrade oscar_django to 2.1, with dependent packages

### PR DESCRIPTION
**Pull request by dianekaplan**
## Description
https://openedx.atlassian.net/browse/REV-2370

Our django-oscar version has been held back at 2.0, and we need to raise it to the next LTS version: 2.1. This includes 4 other package updates, 3 of them major versions (likely used in oscar dashboard if at all): 
- django-haystack (used for search)
- django-phonenumber-field (used for phone numbers)
- django-tables2 (used for creating HTML tables)

## Testing instructions

Please provide detailed step-by-step instructions for testing this change; how did YOU test this change?

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, OpenEdx vs. edx.org differences, development vs. production environment differences, security, or accessibility.

### CODE REVIEW

File: ecommerce/settings/base.py [hunk 1]
```diff
@@ -311,6 +311,11 @@
     'edx_django_utils.user',
 ]
 
+# Apps specific to django-oscar go here.
+OSCAR_APPS = [
+    'oscar.apps.communication.apps.CommunicationConfig',
+    ]
+
 # Apps specific to this project go here.
```
Comment (dianekaplan) [inline ecommerce/settings/base.py]: This is probably wrong- this file is where they exist: 
https://github.com/edx/ecommerce/blob/b69e7e3dce1980782c682e47d9040430fa900754/ecommerce/settings/_oscar.py#L40

File: ecommerce/settings/base.py [hunk 2]
```diff
@@ -311,6 +311,11 @@
     'edx_django_utils.user',
 ]
 
+# Apps specific to django-oscar go here.
+OSCAR_APPS = [
```
Comment (robrap) [inline ecommerce/settings/base.py]: I think this is your issue. You are overwriting: ecommerce/settings/_oscar.py:16. Can you add this (without the class name) in the other file?

Comment (dianekaplan) [reply ecommerce/settings/base.py]: Made the change and will see- thanks!

File: ecommerce/extensions/order/utils.py [hunk 3]
```diff
@@ -114,6 +114,7 @@ def create_order_model(self, user, basket, shipping_address, shipping_method, sh
             order_data['status'] = status
         if extra_order_fields:
             order_data.update(extra_order_fields)
+        Order.surcharges = None
         order = Order(**order_data)
```
Comment (dianekaplan) [inline ecommerce/extensions/order/utils.py:118]: django-oscar 2.1 adds the option to have surcharges. Needed to add this line to keep unit tests passing.

**Comment by dianekaplan** [comment]
Closing in favor of this one: https://github.com/edx/ecommerce/pull/3561/commits

**Pull request closed without merging**